multinet/api/tests/test_workspace.py

```diff
@@ -51,48 +70,295 @@ def test_workspace_rest_create(authenticated_api_client: APIClient):
-    assert not arango_system_db().has_database(owned_workspace.arango_db_name)
+    assert not Workspace.objects.filter(name=workspace.name).exists()
+    assert not arango_system_db().has_database(workspace.arango_db_name)
 
 
 @pytest.mark.django_db
-def test_workspace_rest_delete_unauthorized(owned_workspace: Workspace, api_client: APIClient):
-    r = api_client.delete(f'/api/workspaces/{owned_workspace.name}/')
+def test_workspace_rest_delete_unauthorized(
+    workspace: Workspace, user: User, api_client: APIClient
+):
+    workspace.set_owner(user)
+
+    r = api_client.delete(f'/api/workspaces/{workspace.name}/')
 
     assert r.status_code == 401
 
     # Assert relevant objects are not deleted
-    assert Workspace.objects.filter(name=owned_workspace.name).exists()
-    assert arango_system_db().has_database(owned_workspace.arango_db_name)
+    assert Workspace.objects.filter(name=workspace.name).exists()
+    assert arango_system_db().has_database(workspace.arango_db_name)
 
 
 @pytest.mark.django_db
 def test_workspace_rest_delete_forbidden(
-    workspace_factory: WorkspaceFactory, authenticated_api_client: APIClient
+    workspace: Workspace, user: User, authenticated_api_client: APIClient
+):
+    workspace.set_readers([user])
+    response = authenticated_api_client.delete(f'/api/workspaces/{workspace.name}/')
+    assert response.status_code == 403
+
+    # Assert relevant objects are not deleted
+    assert Workspace.objects.filter(name=workspace.name).exists()
+    assert arango_system_db().has_database(workspace.arango_db_name)
+
+
+@pytest.mark.django_db
+def test_workspace_rest_delete_no_access(
+    private_workspace_factory: PrivateWorkspaceFactory, authenticated_api_client: APIClient
 ):
-    # Create workspace this way, so the authenticated user isn't an owner
-    workspace: Workspace = workspace_factory()
+    # Create workspace this way, so the authenticated user has no access
+    workspace: Workspace = private_workspace_factory()
     r = authenticated_api_client.delete(f'/api/workspaces/{workspace.name}/')
 
-    assert r.status_code == 403
+    # Without any access, don't reveal that this workspace exists at all
+    assert r.status_code == 404
 
     # Assert relevant objects are not deleted
     assert Workspace.objects.filter(name=workspace.name).exists()
     assert arango_system_db().has_database(workspace.arango_db_name)
+
+
+@pytest.mark.django_db
+@pytest.mark.parametrize('permission', [WorkspacePermission.owner, WorkspacePermission.maintainer])
+def test_workspace_rest_get_permissions(
+    workspace: Workspace,
+    user: User,
+    user_factory: UserFactory,
+    authenticated_api_client: APIClient,
+    permission: WorkspacePermission,
+):
+    workspace.set_user_permission(user, permission)
+
+    if permission == WorkspacePermission.maintainer:
+        current_owner = user_factory()
+        workspace.set_owner(current_owner)
```

Comment:
To follow the recommendation of using `set_owner` in the method docstring:
```suggestion
	# Set user perms, ensuring workspace has owner in either case
	if permission == WorkspacePermission.owner:
		workspace.set_owner(user)
	else:
	    workspace.set_owner(user_factory())		
    	workspace.set_user_permission(user, permission)
```